### PR TITLE
Add dependencies to the installation instructions.

### PR DESCRIPTION
**Pull request by ghost**
Expected: installation instructions explain how to install yay.
Actual: `makepkg -si` fails unless the dependencies are installed.

### CODE REVIEW

**Comment by Morganamilo** [comment]
> Before you begin, make sure you have the `base-devel` package group installed.

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ building with makepkg:
 Before you begin, make sure you have the `base-devel` package group installed.
 
 ```sh
+pacman -S git fakeroot binutils gcc make
```
Comment (Jguer) [inline README.md:47]:
```suggestion
pacman -S --needed git base-devel
```

**Comment by Morganamilo** [comment]
If we're going to go that route and add it as an explicit command then you should probably add `--needed` to it.

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md:47]: You can clean up build artifacts by using a multi-stage Dockerfile, however downloading packages that are never used goes against [Lean Flow](https://www.google.com/search?q=lean+flow) principles.  Your batch size and process time will not be as small as possible.  Determine that `git fakeroot binutils gcc make` are the smallest number of dependencies takes time too.

Comment (grandchild) [reply README.md:47]: If having the smallest subset of packages to build yay and _only_ yay is your goal, your argument is reasonable.

However, I would consider another goal, namely that _after_ installing yay, you want to install other AUR packages. And many of those have the implicit assumption that _all_ of base-devel is installed (and won't list them in their dependencies). Failure to instruct users to install base-devel completely will probably result in people coming here or elsewhere to complain about missing package dependencies.

Comment (Jguer) [reply README.md:47]: https://wiki.archlinux.org/index.php/Arch_User_Repository#Getting_started

`Ensure the base-devel package group is installed in full (pacman -S --needed base-devel).`

it's a "minimum requirement" for using the AUR. 

A more experienced user will know how to install with the bare minimum of dependencies. If you want as a compromise you may add a note that to build yay itself only x,y,z are used from base-devel

Comment (grandchild) [reply README.md:47]: > you may add a note that to build yay itself only x,y,z are used from base-devel

But please in an extra section, the installation instructions should be as brief and precise as possible, as not to confuse newcomers.

**Comment by grandchild** [comment]
I'm not so sure about `--needed` after all. Random flags from readme-example lines often wind up copied along into the future because people don't understand why it's there (how could they, they're only just reading on how to get started with this "yay" thing). And yes, omitting `--needed` might re-install `gcc` here and there, but it's a small price to pay for having a really simple installation line, that only has the minimum flags. At least the latter is an appealing feature to me.

**Comment by Morganamilo** [comment]
To be fair there's not really a downside to using `--needed` in any case really.

>Random flags from readme-example lines often wind up copied along into the future because people don't understand why it's there

You're probably right on this but if people don't understand how to use the package manager their distro uses I consider it their fault.

**Comment by grandchild** [comment]
> You're probably right on this but if people don't understand how to use the package manager their distro uses I consider it their fault.

Fair point. And it's not an issue I wanted to press too much. Don't care enough either way :)

**Comment by Jguer** [comment]
Closing due to inactivity

**Pull request closed without merging**